x-pack/plugins/ml/server/shared_services/providers/alerting_service.ts

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { KibanaRequest, SavedObjectsClientContract } from 'kibana/server';
+import { GetGuards } from '../shared_services';
+import { alertingServiceProvider, MlAlertingService } from '../../lib/alerts/alerting_service';
+
+export function getAlertingServiceProvider(getGuards: GetGuards) {
+  return {
+    alertingServiceProvider(
+      savedObjectsClient: SavedObjectsClientContract,
+      request?: KibanaRequest
```

Comment:
Actually, I made it on purpose because `execute` method initializes `fakeRequest`

Reply:
but it isn't `undefined` and on line 20 you're asserting that it isn't `undefined`

Reply:
i see, i missed that `fakeRequest` was being created in the `execute` function.
I wonder if this will cause problems for people using this for the `preview` function, as the types appear that `request` is optional.

Reply:
perhaps the `fakeRequest` should be created in the `executor` which uses this?
this would make it consistent with all of the other shared service providers.
And hopefully alerts remove the need for us to create these fake requests entirely soon.

Reply:
changed in https://github.com/elastic/kibana/pull/89286/commits/7a320cce1da122668678f282b053dc2dceac7dfe